File: src/tests/Mappings/BookAuthors.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace Extender
{

      [Serializable()]
		public class BookAuthors : TableMetadata
		{

                   public enum BookAuthorsFields
                   {
                      Id,
                      AuthorId,
                      BookId
                  }


			    private DatabaseField[] _fields;

		    	public BookAuthors()
			    {
					    _fields = new DatabaseField[3];
                    _fields[0] = new DatabaseField(DbType.Int32,"Id",true,true,null);
                    _fields[1] = new DatabaseField(DbType.Int32,"AuthorId",false,false,null);
                    _fields[2] = new DatabaseField(DbType.Int32,"BookId",false,false,null);
 
                        this.currentTableName = "BookAuthors";

this.listRelations.Add(new ChildTableRelation("Author",TableRelationCardinality.OneToOne,"AuthorId","AuthorId"));

                  }


			public override DatabaseField[] TableFields 
			{
				get{ return _fields;}
				set{_fields = value;}
			}
          public BookAuthors Clone()
          {
                 return this.Clone<BookAuthors>();
          }

public System.Int32? Id
{
    get
    {
          return (System.Int32? ) (this.GetField("Id")).fieldValue;
    }

    set
    {
          this.SetFieldValue("Id", value);
    }
}


public System.Int32? AuthorId
{
    get
    {
          return (System.Int32? ) (this.GetField("AuthorId")).fieldValue;
    }

    set
    {
          this.SetFieldValue("AuthorId", value);
    }
}


public System.Int32? BookId
{
    get
    {
          return (System.Int32? ) (this.GetField("BookId")).fieldValue;
    }

    set
    {
          this.SetFieldValue("BookId", value);
    }
}

public TableMetadata GetAuthor()
{
    Type relatedType = Type.GetType("Extender.Author");
    TableMetadata[] result = this.GetRelatedTableData(relatedType);
    return result.Length > 0 ? result[0] : null; 
}
}
}

File: src/tests/MappingsVB/BookAuthors.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock

Namespace Extender
 


		Public Class BookAuthors
		             Inherits TableMetadata
			Private _fields() as DatabaseField

			Sub New() 
			
					_fields = new DatabaseField(2) {}
 _fields(0) = new DatabaseField(DbType.Int32,"Id",True,True,Nothing)
 _fields(1) = new DatabaseField(DbType.Int32,"AuthorId",False,False,Nothing)
 _fields(2) = new DatabaseField(DbType.Int32,"BookId",False,False,Nothing)
 
me.currentTableName = "BookAuthors"


End Sub



			Public Overrides Property TableFields() as DatabaseField() 
		   	   Get
				  Return _fields
			   End Get
			   Set (ByVal Value as DatabaseField())
			      _fields = value
			   End Set
			End Property

     Public Property Id As System.Int32
        Get
           Return  CType((Me.GetField("Id")).fieldValue,System.Int32)
        End Get 
        Set(ByVal Value As System.Int32)
          Me.SetFieldValue("Id", Value)
        End Set
      End Property

     Public Property AuthorId As System.Int32
        Get
           Return  CType((Me.GetField("AuthorId")).fieldValue,System.Int32)
        End Get 
        Set(ByVal Value As System.Int32)
          Me.SetFieldValue("AuthorId", Value)
        End Set
      End Property

     Public Property BookId As System.Int32
        Get
           Return  CType((Me.GetField("BookId")).fieldValue,System.Int32)
        End Get 
        Set(ByVal Value As System.Int32)
          Me.SetFieldValue("BookId", Value)
        End Set
      End Property
End Class
End Namespace
